Phase03

Melikad opened the pull request:
unit testing

Melikad committed:
Main codes transfered to src folder.

rahaahmadi committed:
Update

rahaahmadi committed:
Update

rahaahmadi committed:
Bug fixed

rahaahmadi committed:
Word wrap deleted

rahaahmadi committed:
Used built in method in getWordDocs method.

rahaahmadi committed:
Update DocsFileReader.java

rahaahmadi committed:
Change method visibility

rahaahmadi committed:
Change test string

Melikad committed:
method name corrected

rahaahmadi committed:
Refactor InvertedIndexTest.java

rahaahmadi committed:
Added test for Filters

rahaahmadi committed:
Fixed bug

rahaahmadi committed:
Update

rahaahmadi committed:
Update FullTextSearchTest.java

zkhotanlou commented on Phase1/test/FilterHandlerTest.java:
```diff
@@ -0,0 +1,29 @@
+import static org.junit.Assert.*;
+
+public class FilterHandlerTest {
+    private InvertedIndex invertedIndex = new InvertedIndex();
```
When your are using methods from other classes (like filter method in plusFilter, minusFilter, etc..) you should mock the object and stub these methods. for example imagine some situation that filter method from PlusFilter doesn't work well so this test will be failed too and make debugging code annoying for you.

zkhotanlou commented on Phase1/test/FullTextSearchTest.java:
```diff
@@ -0,0 +1,49 @@
+    private NoSignFilter noSignFilter;
+
+    @Before
+    public void initialize() {
```
setup name for this method is more common

zkhotanlou commented on Phase1/test/FullTextSearchTest.java:
```diff
@@ -0,0 +1,49 @@
+import org.junit.Before;
+
+public class FullTextSearchTest {
+    private InvertedIndex invertedIndex;
```
like previous comment, sounds that you should mock these object

nedataghizadeh79 commented on Phase1/test/FilterHandlerTest.java:
```diff
@@ -0,0 +1,29 @@
+    public void testFilter() {
+        initializeInvertedIndex();
+        FilterHandler filterHandler = new FilterHandler(invertedIndex, plusFilter, noSignFilter, minusFilter);
+        String[] commandWords = { "+s1", "-s2", "s6" };
```
bhtar bood halate digar niz test shavad baraye mesal :
faghat "+s1"
ya 
faghat "-s2"
:)

hoseinaghaei replied:
به این سبک از تست نوشتن که حالت های مختلف ورودی چک شود 
Integration test
میگن و در پروژه هایی مثل اینجا که الگوریتمی هستند بهتره انجام شه

zkhotanlou commented on Phase1/test/FullTextSearchTest.java:
```diff
@@ -0,0 +1,49 @@
+    @Test
+    public void testFullTextSearch() {
+        initialize();
+        // ArrayList<String> tests = new ArrayList<>(
```
unnecessary comments :)

zkhotanlou commented on Phase1/test/FullTextSearchTest.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Test
+    public void testFullTextSearch() {
+        initialize();
```
implement this method with @Before annotation cause that call it automatically before run test so this line is useless

rahaahmadi committed:
Use mock

rahaahmadi committed:
Merge remote-tracking branch 'origin/Phase03' into Phase03

rahaahmadi committed:
Revert to previous version

rahaahmadi committed:
Updated

hoseinaghaei commented on Phase1/src/DocsFileReader.java:
```diff
@@ -8,17 +10,17 @@ public DocsFileReader() {
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
             }
+        } catch (Exception e) {
```
باید نوع اکسپشن رو مشخص کنید  و سعی کنید پیام قابل فهم بدین براش

hoseinaghaei commented on Phase1/src/FullTextSearch.java:
```diff
@@ -16,8 +18,8 @@ public FullTextSearch(String command, InvertedIndex invertedIndex, Tokenizer tok
-    public void loadDocs() {
-        final String fileAddress = "Phase1/EnglishData";
+    private void loadDocs() {
+        final String fileAddress = "C:/Users/ASUS/Desktop/Summer1400-SE-Team13/Summer1400-SE-Team13/Phase1/test/SampleFolder";
```
Absolute path❌❌

hoseinaghaei commented on Phase1/src/InvertedIndex.java:
```diff
@@ -27,15 +29,10 @@ private void addWord(String word) {
-        if (containsWord(word))
-            return wordsMap.get(word);
-        return null;
+        return wordsMap.getOrDefault(word, null);
```
از تابع 
getOrDefault
وقتی استفاده میکنیم که اگر هش مپ شامل کی نبود یک چیز خاصی برگردونیم 
شما دارین نال میدین در حالی که تابع 
get 
اگر کی نباشه نال میده، نیاز به این تابع نیست دیگه اینطوری

hoseinaghaei commented on Phase1/src/NoSignFilter.java:
```diff
@@ -1,8 +1,12 @@
         if (docs != null)
             result.retainAll(docs);
+        else
+            result.clear();
```
اگر تو تابع 
getOrDefault
بجای نال، یک هش ست خالی بدین، اینجا دیگه نیاز به شرط نال چک کردن هم نیست

hoseinaghaei commented on Phase1/test/DocsFileReaderTest.java:
```diff
@@ -0,0 +1,21 @@
+    public void testFileReader() {
+        DocsFileReader docsFileReader = new DocsFileReader();
+        String docContentExpected = "Hello World.";
+        Integer docIDExpected = 1;
```
primitive
تعریف کنین

hoseinaghaei commented on Phase1/test/DocsFileReaderTest.java:
```diff
@@ -0,0 +1,21 @@
+        String docContentExpected = "Hello World.";
+        Integer docIDExpected = 1;
+        HashMap<Integer, String> expectedValue = new HashMap<>();
+        expectedValue.put(docIDExpected, docContentExpected);
```
اماده کردن 
expected value
رو میتونین به یک تابع دیگه بسپپارین

hoseinaghaei commented on Phase1/test/NoSignFilterTest.java:
```diff
@@ -0,0 +1,19 @@
+        HashSet<Integer> res = new HashSet<>(Arrays.asList(1, 2, 3, 4, 5));
+        HashSet<Integer> docs = new HashSet<>(Arrays.asList(4, 5));
+        noSignFilter.filter(res, docs);
+        assertEquals(expectedResult, res);
```
اینکه کدتون بهم کاپل شده رو تو فاز ۵ از بین ببرین

hoseinaghaei commented on Phase1/test/TokenizerTest.java:
```diff
@@ -0,0 +1,17 @@
+    @Test
+    public void testTokenizer() {
+        HashSet<String> expected = new HashSet<>(Arrays.asList("hello","hi", "java", "python", "how", "test"));
+        String str = "Hello-hi, how? java*python! test.";
```
اسم درست بذارین

hoseinaghaei reviewed (commented):
مشکل بزرگی نبود اما کامنت هارو حتما برین رفع کنین

Parsa2820 commented on Phase1/test/InvertedIndexTest.java:
```diff
@@ -0,0 +1,38 @@
+    private static final HashMap<String, HashSet<Integer>> testWordsMap = new HashMap<>();
+
+    private void initWordsMap() {
+        HashSet<Integer> h1 = new HashSet<>(Collections.singletonList(1));
```
اسم مرتبط بذارید

Parsa2820 reviewed (commented):
خوب بود ممنون. تو سی شارپ حتما حتما از ماک استفاده کنید. خسته نباشید

Melikad committed:
exception type added

Melikad committed:
integrated tests added to FullTextSearchTest

Melikad committed:
DocsFileReaderTest expected values transferred to another new method
Integer declared in primitive format (docsFileReaderTest)
getOrDefault bugs fixed in InvertedIndex and filter classesI
some variable naming changed

Melikad committed:
changed some namings
